mlflow/tracking/fluent.py

```diff
@@ -2372,6 +2378,32 @@ def print_auto_logged_info(r):
         # do not enable langchain autologging by default
     }
 
+    GENAI_LIBRARY_TO_AUTOLOG_MODULE = {
+        "anthropic": "mlflow.anthropic",
+        "autogen": "mlflow.autogen",
+        "openai": "mlflow.openai",
+        "google.generativeai": "mlflow.gemini",
+        "litellm": "mlflow.litellm",
+        "llama_index.core": "mlflow.llama_index",
+        "langchain": "mlflow.langchain",
+        "dspy": "mlflow.dspy",
+    }
+
+    # Currently, GenAI libraries are not enabled by `mlflow.autolog` in Databricks,
+    # particularly when disable=False. This is because the function is automatically invoked
+    # by system and we don't want to take the risk of enabling GenAI libraries all at once.
+    # TODO: Remove this logic once a feature flag is implemented in Databricks Runtime init logic.
+    if is_in_databricks_runtime() and (not disable):
+        target_library_and_module = LIBRARY_TO_AUTOLOG_MODULE
+    else:
+        target_library_and_module = LIBRARY_TO_AUTOLOG_MODULE | GENAI_LIBRARY_TO_AUTOLOG_MODULE
+
+    if exclude_flavors:
+        excluded_modules = [f"mlflow.{flavor}" for flavor in exclude_flavors]
+        target_library_and_module = {
+            k: v for k, v in LIBRARY_TO_AUTOLOG_MODULE.items() if v not in excluded_modules
+        }
```

Comment:
```suggestion
        target_library_and_module = {
            k: v for k, v in target_library_and_module.items() if v not in excluded_modules
        }
```

should we use `target_library_and_module` here?

Reply:
Good catch, thank you!